ruby-challenges.rb

```diff
@@ -10,22 +10,68 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def letter_array (array, letter)
+    array.select {|value| value.include?(letter)}
+    end
+
+
+ letter_array(beverages_array, letter_o)
+ letter_array(beverages_array, letter_t)
+
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+def state_list obj
+    array_storage = obj.values
+    ordered_arr = array_storage[0].concat(array_storage[1].concat(array_storage[2])).sort()
+end
+
+state_list us_states
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike
+    def initialize(model, wheels)
+        @model = model
+        @wheels = wheels
+        @current_speed = 0
+    end
+    
+    def bike_info
+        p "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end 
+    
+    def pedal_faster speed
+        @current_speed += speed
+    end
+
+    def brake speed
+        if @current_speed - speed < 0
+            @current_speed = 0
+        else 
+            @current_speed -= speed
+        end    
+    end
+
+end
+my_bike = Bike.new("Trek", 2)
+
```

Comment:
If you needed to reassign a value to any of your instance variables, what would you add to your class?